pkg/parse/root_test.go

```diff
@@ -451,6 +454,221 @@ func TestRoot_Parse(t *testing.T) {
 	}
 }
 
+func fakeCRD(opts ...core.MetaMutator) ast.FileObject {
+	//opts = append(opts, core.Name("anvils.acme.com"))
+	crd := fake.CustomResourceDefinitionV1Object(opts...)
+	crd.Spec.Group = "acme.com"
+	crd.Spec.Names = apiextensionsv1.CustomResourceDefinitionNames{
+		Plural:   "anvils",
+		Singular: "anvil",
+		Kind:     "Anvil",
+	}
+	crd.Spec.Scope = apiextensionsv1.ResourceScope(discovery.UnknownScope)
+	crd.Spec.Versions = []apiextensionsv1.CustomResourceDefinitionVersion{
+		{
+			Name:    "v1",
+			Served:  true,
+			Storage: false,
+			Schema: &apiextensionsv1.CustomResourceValidation{
+				OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+					Type: "object",
+					Properties: map[string]apiextensionsv1.JSONSchemaProps{
+						"spec": {
+							Type:     "object",
+							Required: []string{"lbs"},
+							Properties: map[string]apiextensionsv1.JSONSchemaProps{
+								"lbs": {
+									Type:    "integer",
+									Minimum: pointer.Float64Ptr(1.0),
+									Maximum: pointer.Float64Ptr(9000.0),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return fake.FileObject(crd, "cluster/crd.yaml")
+}
+
+func TestRoot_Parse_Discovery(t *testing.T) {
```

Comment:
Can you also add a test that fakes a timeout error from the discovery client? And one that shows how it handles multiple different errors? Maybe a 500 error?

Reply:
Done, added two error types: context deadline exceeded and net/http timeout error. Also added a case with 500 errors. PTAL